pkg/netns/netns_linux.go

```diff
@@ -28,15 +28,25 @@ import (
 	"sync"
 
 	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containers/libpod/pkg/rootless"
 	"golang.org/x/sys/unix"
 )
 
-const nsRunDir = "/var/run/netns"
+// get NSRunDir returns the dir of where to create the netNS. When running
+// rootless, it needs to be at a location writable by user.
+func getNSRunDir() string {
+	if rootless.IsRootless() {
+		return "/tmp/netns"
+	}
+	return "/var/run/netns"
+}
 
 // NewNS creates a new persistent (bind-mounted) network namespace and returns
 // an object representing that namespace, without switching to it.
 func NewNS() (ns.NetNS, error) {
```

Comment:
should we also drop the check in `createNetNS` where we error out immediately when running in rootless mode?

Reply:
That's a good idea, and then have some sort of logic for "`configureNetNS`" to be to setup the networking depending on the privilege (CNI/slirp/etc)?